src/cpp/src/group_beam_searcher.cpp

```diff
@@ -452,9 +449,10 @@ EncodedResults beam_search(ov::InferRequest& lm,
         ) {
             results.scores.push_back(beam->get().score);
             results.tokens.push_back(std::move(beam->get().tokens));
+            selected_beams.push_back(beam->get().global_beam_idx);
```

Comment:
`selected_beam_idx.size()` is expected to be `1` because every beam is spawned from a single prompt. But `selected_beams` can be pushed to multiple times at this line. So the next `beam_search()` will fail at `OPENVINO_ASSERT((*selected_beam_idx).size() == 1);` for chat.

`selected_beam_idx` shouldn't be a vector. It's just `std::optional<int32_t>`.

Is the way to propagate an accepted by the user beam missing for chat? (https://github.com/openvinotoolkit/openvino.genai/pull/549/files#r1654606288 is probably the same question)

Reply:
>selected_beam_idx shouldn't be a vector. It's just std::optional<int32_t>.

done